dftimewolf/lib/processors/workspace_audit_timesketch.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+"""Processes Google Workspace logs for loading into Timesketch."""
+
+import os
+import tempfile
+import json
+import string
+
+from dftimewolf.lib.module import BaseModule
+from dftimewolf.lib.containers import containers
+
+from dftimewolf.lib.modules import manager as modules_manager
+
+
+class WorkspaceAuditTimesketch(BaseModule):
+  """Transforms Google Workspace logs for Timesketch."""
+
+  _FORMAT_STRINGS_PATH = os.path.join(os.path.dirname(__file__),
+      'workspace_format_strings.json')
+
+  def __init__(self, state, name=None, critical=False):
+    super(WorkspaceAuditTimesketch, self).__init__(
+        state, name=name, critical=critical)
+
+    with open(self._FORMAT_STRINGS_PATH, 'r') as formatters_json:
+      self._all_application_format_strings = json.load(formatters_json)
+
+  def SetUp(self, *args, **kwargs):
+    """Sets up necessary module configuration options."""
+    # No configuration required.
+
+  def _FlattenActor(self, actor_dict):
+    """Flattens out actor information from a Workspace log record.
+
+    Args:
+      actor_dict (dict): contents of the 'actor' dict in a Workspace log record
+
+    Returns:
+      dict[str, str]: a dictionary containing actor information suitable for
+          adding to a Timesketch record.
+    """
+    return {
+        'actor_email': actor_dict.get('email'),
+        'actor_profileId': actor_dict.get('profileId'),
+        'actor_callerType': actor_dict.get('callerType'),
+        'actor_key': actor_dict.get('key')}
+
+  def _FlattenParameters(self, parameters):
+    """Flattens out parameter information from a Workspace log record.
+
+    The parameter list looks like this:
+      [{"name": "event_id", "value": "4"}, {"name": "title", "value": "foo"}]
+
+    This method turns it into:
+      {"event_id": 4, "title": "foo"}
+
+    Args:
+      parameters (list): the contents of a Workspace parameters list.
+
+    Returns:
+      dict[str, str]: a dictionary containing parameter information suitable for
```

Comment:
nit; the example right above has an `int` as value.

Reply:
Done